File: sdks/ruby/lib/novabilling/types/ltv_response.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class LtvResponse < Internal::Types::Model
      field :avg_ltv, -> { Integer }, optional: false, nullable: false, api_name: "avgLtv"
      field :avg_lifespan_days, -> { Integer }, optional: false, nullable: false, api_name: "avgLifespanDays"
      field :by_plan, -> { Internal::Types::Array[Novabilling::Types::LtvPlanBreakdown] }, optional: false, nullable: false, api_name: "byPlan"
    end
  end
end

File: sdks/csharp/src/NovaBilling/Types/LtvResponse.cs
using System.Text.Json;
using System.Text.Json.Serialization;
using NovaBilling.Core;

namespace NovaBilling;

[Serializable]
public record LtvResponse : IJsonOnDeserialized
{
    [JsonExtensionData]
    private readonly IDictionary<string, JsonElement> _extensionData =
        new Dictionary<string, JsonElement>();

    [JsonPropertyName("avgLtv")]
    public required double AvgLtv { get; set; }

    [JsonPropertyName("avgLifespanDays")]
    public required double AvgLifespanDays { get; set; }

    [JsonPropertyName("byPlan")]
    public IEnumerable<LtvPlanBreakdown> ByPlan { get; set; } = new List<LtvPlanBreakdown>();

    [JsonIgnore]
    public ReadOnlyAdditionalProperties AdditionalProperties { get; private set; } = new();

    void IJsonOnDeserialized.OnDeserialized() =>
        AdditionalProperties.CopyFromExtensionData(_extensionData);

    /// <inheritdoc />
    public override string ToString()
    {
        return JsonUtils.Serialize(this);
    }
}
